### ISSUE TITLE
Uri.EscapeUriString() not escaping Form Json Apostrophe on Script.cshtml

### ISSUE THREAD

**Issue by sattapaal** [umbraco/Umbraco.Forms.Issues#456]
Hi,

I have been working with Umbraco Forms 8.2 then converted it to 8.5.5 (umbraco version 8.8.1). On the new Scripts.cshtml within Views/Forms/Partials/ this line seems to break when a field has an apostrophe in it:

`umbracoFormsCollection.push('@Html.Raw(Uri.EscapeUriString(JsonConvert.SerializeObject(formJsObj)))');`

a tester found that if they input an apostrophe, it breaks the conditions. but its much more severe than that. looking at the logs the following is there:

`System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Umbraco.Forms.Core.Data.Storage.RecordFieldStorage.GetAllRecordFields(Record record, Form form)
   at Umbraco.Forms.Core.Data.Storage.RecordStorage.GetRecordByUniqueId(Guid uniqueId, Form form)
   at Umbraco.Forms.Web.Controllers.UmbracoFormsController.GetRecord(Guid recordId, Form form)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at LGO.Controllers.FormsController.BaseGetRecord(Guid recordId, Form form)`

at the moment, I have resolved it by using the following:

`umbracoFormsCollection.push("@(Html.Raw(Uri.EscapeUriString(JsonConvert.SerializeObject(formJsObj))))");`

double quotes seem to work fine when inputting them into a field....but i think this is largely due to the JsonConvert. I dont really know what the Uri.EscapeUriString does, but for me, it doesnt seem to escape single quotes. 

Please note, this is a multi page form, and i've had to use system reflection to "skip" over steps that do not show any fields + other things that manipulate the FormState.

Regards,
Satpal

---
_This item has been added to our backlog [AB#9751](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/9751)_

**Comment by nul800sebastiaan** [comment]
Ah yes, your solution is actually perfect. When adding a single quote in javascript surrounded by single quotes.. well it ends the string. 

I then thought: oh no.. if you test with entering double quotes then it's going to break using your solution, but luckily, `EscapeUriString` already encodes a double quote properly! 👍

**Comment by saadan** [comment]
Think I walked into this error in Umbraco Forms: 8.6.0 
I was making a condition where a input field must only be visible when a dropdown is 'text, 

First I got this error. 
![image](https://user-images.githubusercontent.com/1712539/122743123-fd2d7f00-d286-11eb-9e4b-cb656e61e28a.png)
This was fixed with the double quotes.

but unfortunately now I gut this error when selecting the dropdown value, so it's not resolved completely.
![image](https://user-images.githubusercontent.com/1712539/122742242-079b4900-d286-11eb-9260-fc17c7b9296d.png)
